dist/categorical.go

```diff
@@ -91,10 +90,7 @@ func (c Categorical) Mean() float64 {
 // Prob computes the value of the probability density function at x.
 func (c Categorical) Prob(x float64) float64 {
 	xi := int(x)
-	if float64(xi) != x {
```

Comment:
While this is logically identical, the reader-facing semantics are different for the two conditions and so are separate.